src/cmds/net/telnetd/telnetd.c

```diff
@@ -48,23 +45,22 @@
 	} while (0);
 #endif
 
-#define T_WILL		251
-#define T_WONT		252
-#define T_DO		253
-#define T_DONT		254
-#define T_IAC		255
+/* http://www.tcpipguide.com/free/t_TelnetProtocolCommands-3.htm */
+#define T_WILL      251
+#define T_WONT      252
+#define T_DO        253
+#define T_DONT      254
+#define T_IAC       255
 #define T_INTERRUPT 244
 
-#define O_ECHO		1		/* Manage ECHO, RFC 857 */
-#define O_GO_AHEAD	3		/* Disable GO AHEAD, RFC 858 */
+#define O_ECHO      1     /* Manage ECHO, RFC 857 */
+#define O_GO_AHEAD  3     /* Disable GO AHEAD, RFC 858 */
 
+//TODO is not posix. Must be posix_openpt() and friends
 extern int ppty(int pptyfds[2]);
 
-static struct {
-	int fd;
-	struct sockaddr_in addr_in;
-} clients[TELNETD_MAX_CONNECTIONS];
-static int listening_descr;
+
+static int telnet_connections_count;
```

Comment:
As far as I understood, race condition could make telnet refuse connections. Why it's not significant? In some time system will be unreachable by telnet.
It also allows to bypass connection limit, accepting more connections than allowed. Since our usual configurations are strictly limited by number of threads/tasks, this could make the system to fail another thread/task creation.